search/search_quality_tests/search_quality_tests.cpp

```diff
@@ -0,0 +1,153 @@
+  {
+    return make_unique<search::v2::SearchQueryV2>(index, categories, suggests, infoGetter);
+  }
+};
+
+void ReadStringsFromFile(string const & path, vector<string> & result)
+{
+  ifstream stream(path.c_str());
+  // Using errno here gives no guarantees but is better than nothing.
+  CHECK(stream.is_open(), ("Error when reading strings from", path, ":", string(strerror(errno))));
+  MY_SCOPE_GUARD(cleanup, [&]
+                 {
+                   stream.close();
+                 });
+
+  string s;
+  while (getline(stream, s))
+  {
+    strings::Trim(s);
+    if (!s.empty())
+      result.emplace_back(s);
+  }
+}
+
+// If n == 1, prints the query and the top result separated by a tab.
+// Otherwise, prints the query on a separate line
+// and then prints n top results on n lines starting with tabs.
+void PrintTopResults(string const & query, vector<search::Result> const & results, size_t n)
+{
+  cout << query;
+  for (size_t i = 0; i < n; ++i)
+  {
+    if (n > 1)
+      cout << "\n";
+    cout << "\t";
+    if (i < results.size())
+      // todo(@m) Print more information: coordinates, viewport, etc.
+      cout << string(results[i].GetString());
+    else
+      cout << kEmptyResult;
+  }
+  cout << "\n";
+}
+
+int main(int argc, char * argv[])
+{
+  ios_base::sync_with_stdio(false);
+  Platform & platform = GetPlatform();
+
+  google::SetUsageMessage("Search quality tests.");
+  google::ParseCommandLineFlags(&argc, &argv, true);
+
+  if (!FLAGS_data_path.empty())
+    platform.SetResourceDir(FLAGS_data_path);
+
+  if (!FLAGS_mwm_path.empty())
+    platform.SetWritableDirForTests(FLAGS_mwm_path);
+
+  LOG(LINFO, ("writable dir =", platform.WritableDir()));
+  LOG(LINFO, ("resources dir =", platform.ResourcesDir()));
+
+  classificator::Load();
+
+  TestSearchEngine engine(FLAGS_locale, make_unique<SearchQueryV2Factory>());
+
+  vector<platform::LocalCountryFile> mwms;
+  if (!FLAGS_mwm_list_path.empty())
+  {
+    vector<string> availableMwms;
+    ReadStringsFromFile(FLAGS_mwm_list_path, availableMwms);
+    for (auto const & countryName : availableMwms)
+      mwms.emplace_back(platform.WritableDir(), platform::CountryFile(countryName), 0);
+  }
+  else
+  {
+    platform::FindAllLocalMapsAndCleanup(numeric_limits<int64_t>::max() /* the latest version */,
+                                         mwms);
+    for (auto & map : mwms)
+      map.SyncWithDisk();
+  }
+  for (auto const & mwm : mwms)
+  {
+    LOG(LINFO, ("Registering map:", mwm.GetCountryName()));
+    engine.RegisterMap(mwm);
+  }
+
+  vector<string> queries;
+  string queriesPath = FLAGS_queries_path;
+  if (queriesPath.empty())
+    queriesPath = my::JoinFoldersToPath(platform.WritableDir(), kDefaultQueriesPathSuffix);
+  ReadStringsFromFile(queriesPath, queries);
+
+  for (string const & query : queries)
+  {
+    // todo(@m) Viewport and position should belong to the query info.
```

Comment:
Probably it's worth to pass them through command line. It's up to you whether it worth to implement it right now or delay until we will need custom viewport and position.